sway-core/src/lib.rs

```diff
@@ -376,6 +376,34 @@ pub(crate) fn compile_ast_to_ir_to_asm(
     let mut warnings = Vec::new();
     let mut errors = Vec::new();
 
+    // If the program is a script or a predicate,
+    // check if its main() returns a type containing a raw_ptr.
+    match &program.kind {
```

Comment:
Can we do this check in https://github.com/FuelLabs/sway/blob/c096c4a640319ac020e4f20baa898e081f6825fb/sway-core/src/semantic_analysis/program.rs#L61? We have a bunch of checks there that seems quite similar.

Reply:
Also, predicates need to return a `bool` anyways so hopefully that's checked somewhere and doesn't need to be checked again.